editor/scene_tree_dock.cpp

```diff
@@ -185,21 +185,33 @@ void SceneTreeDock::_perform_instance_scenes(const Vector<String> &p_files, Node
 		}
 
 		if (edited_scene->get_filename() != "") {
-
-			if (_cyclical_dependency_exists(edited_scene->get_filename(), instanced_scene)) {
-
-				accept->set_text(vformat(TTR("Cannot instance the scene '%s' because the current scene exists within one of its nodes."), p_files[i]));
-				accept->popup_centered_minsize();
-				error = true;
-				break;
+			Node *p = instanced_scene;
+			while (p) {
+				if (_cyclical_dependency_exists(edited_scene->get_filename(), p)) {
+					accept->set_text(vformat(TTR("Cannot instance the scene '%s' because the current scene exists within one of its nodes."), p_files[i]));
+					accept->popup_centered_minsize();
+					error = true;
+					memdelete(p);
+					goto Skip;
```

Comment:
Can't this be an else branch instead?

Reply:
It will track a the inherted scene's parent scene, so i use loop to track upwards here.  I found some other questions from those code, it's hard to check the scene's relationship when use inhert and instance at the same time, I'm trying to modify it